polling_stations/apps/data_finder/helpers.py

```diff
@@ -191,55 +191,67 @@ def geocode_point_only(self):
 
 
 def geocode_point_only(postcode, sleep=True):
-    addressbase = AddressBaseGeocoder(postcode)
-    mapit = MapitGeocoder(postcode)
-
-    try:
-        # first try addressbase
-        result = addressbase.geocode_point_only()
-    except ObjectDoesNotExist:
-        # we couldn't find this postcode in AddressBase: fall back to mapit
+    geocoders = (AddressBaseGeocoder(postcode), MapitGeocoder(postcode))
+    for geocoder in geocoders:
+        try:
+            return geocoder.run(True)
+        except ObjectDoesNotExist:
+            # we couldn't find this postcode in AddressBase
+            # fall back to the next source
 
-        # optional sleep to avoid hammering mapit
-        if sleep:
-            time.sleep(1.3)
+            # optional sleep to avoid hammering external services
+            if sleep:
```

Comment:
Just a small thing: is it worth passing `sleep` on to the classes and letting them sleep if they need to? Otherwise we might end up in the situation where we sleep in between AddressBase calls. Basically, each class should be able to ignore sleep if it wants to.